### ISSUE TITLE
Approach for incorporating external samples

### ISSUE THREAD

**Issue by tomkerkhove** [kedacore/keda#629]
In last standup the following was identified:

> Should have a way for folks to have their Samples in the KEDA repo. Anirudh to follow up 

We should provide and document an approach for external samples.

**Comment by tomkerkhove** [comment]
@rasavant-ms @anirudhgarg Was anything more than that discussed?

I see two types of samples:
- The ones we maintain today, or are happy to take over and maintain
- External ones provided by folks in the community

Would you want to open it up for everbody to move their repo in our org? If so, how will we keep them all up to date?

What if we use https://github.com/kedacore/samples as an incubation where folks can PR their sample in. If folks are interested, they can log an issue there saying they want to donate it and we can evaluate it so that:
1. We don't already have a similar sample
2. We have bandwidth to keep it up to date

The reason why I'm skeptical is that we don't perse have to own all the repos as others have done the hard work, it's less work for us to keepe them up to date and our org would become huge.

**Comment by anirudhgarg** [comment]
@tomkerkhove thanks for chiming in. 
Currently we have the following ways we are handling samples: 
1). We have a samples repo which is basically a link file (linking to different samples)
2). Repos within KEDA org where every repo is a sample

The ask was that if we could also have Samples within the Samples repo (instead of creating a whole repo for a Sample). I think it makes sense as you too pointed out that we should allow folks to submit PR's to the Samples repo and add their Samples there as to not increase the number of repos in the org itself. 

Looks like you agree to this - so @rasavant-ms  feel free to submit a PR on the Samples repo with your sample - we can take this as a test to see how it works out. 




Basically the ask was that if folks can also use (if they want to) the samples repo to host their samples themselves - currently its a single page with links to samples.

**Comment by tomkerkhove** [comment]
Ah so your idea is to host the content in that repo as well then? What is the reasoning behind that?

Wouldn't that become a big pile of sample dump over time?

Let me ask it otherwise, why does Azure have a repo per sample for example and not one repo with all samples in it?

My suggestion with the PR was more to include a like to their repo, or file an issue as a proposal to donate.

The risk I see is that we end up with a ton of samples which are no longer maintained which is bad from a product perspective.

**Comment by marchmallow** [comment]
Why not allow people to submit a PR to link their own repos as samples into the sample repo? And divide the list between internally maintained samples and community/externally provided ones? I was thinking we could submit an example of what we are doing with redis and this would work easy enough for me..

**Comment by tomkerkhove** [comment]
We have https://github.com/kedacore/samples nowadays, closing
